### PR TITLE
refactor: @masknet/gun-utils package

### PR DESCRIPTION
**Pull request by Jack-Works**
-   feat: add gun-utils package
-   chore: add gun-utils as dep
-   chore: migrate appendShareTarget
-   chore: remove debugShowAllPossibleHashForPost
-   chore: migrate encryptTo
-   chore: migrate getPartialSharedListOfPost
-   chore: migrate avatar service
-   chore: migrate poll plugin
-   chore: migrate decryptFrom
-   chore: remove old gun code
-   chore: remove usage of ondemandworker in poll

## Description

This PR is picked from #5208 which is too big.

This PR introduce a new package `@masknet/gun-utils` to simplify the usage with gun.

It provides a convenient surface for accessing gun. It exports the following APIs. All access to the Gun must use this
package.

Provided a self-destructing WebSocket constructor to the Gun therefore OnDemandWorker is no longer required for Gun

```ts
// Normal access
deleteGunData
getGunData
setGunData

// Map access
pushToGunDataArray

// Subscribe
subscribeGunMapData
```

Closes # (NO_ISSUE)

## Type of change

-   [ ] Documentation
-   [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must
        upgrade their Mask Network to this new version)

## Checklist

-   [x] My code follows the style guidelines of this project.
-   [x] I have performed a self-review of my own code.
    -   [x] I have removed all in development `console.log`s
    -   [x] I have removed all commented code.
-   [x] I have commented on my code, particularly in hard-to-understand areas.
-   [x] I have read
        [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved
        text fields to the i18n JSON file.

If this PR depends on external APIs:

-   [ ] I have configured those APIs with CORS headers to let extension requests get passed.
        
    -   chrome extension: `chrome-extension://[id]`
    -   firefox extension: `moz-extension://[id]`
-   [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: pnpm-workspace.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ packages:
   - 'packages/dashboard'
   - 'packages/mask'
   - 'packages/injected-script'
+  - 'packages/gun-utils'
```
Comment (UncleBill) [inline pnpm-workspace.yaml:12]: why don't we use glob pattern here?

```yml
packages:
  - 'packages/*'
  - 'packages/polyfills/*'
 ```

Comment (Jack-Works) [reply pnpm-workspace.yaml:12]: More explicit is good? 🤔

File: packages/mask/src/plugins/Polls/Services.ts [hunk 2]
```diff
@@ -101,59 +94,51 @@ export async function vote(props: voteProps) {
...
-        key: props.key,
         id: first(keys),
+        key: props.key,
+        sender: data.sender,
+        question: data.question,
+        start_time: data.start_time,
+        end_time: data.end_time,
+    }
+    if (data.options) {
+        const options = await getGunData(PollGunRootNode, props.key, 'options')
```
Comment (septs) [inline packages/mask/src/plugins/Polls/Services.ts:111]: i think can provide a gundb agent object?

```
const gun = new GunAgent(/** prefix */ PollGunRootNode)

gun.get(...)
gun.set(...)
gun.delete(...)
// etc
```

Comment (Jack-Works) [reply packages/mask/src/plugins/Polls/Services.ts:111]: I explicitly limit the complexity of this wrapper package to make it easy to use.

File: pnpm-workspace.yaml (hunk 1 above)
Comment (UncleBill) [reply pnpm-workspace.yaml:12]: I think we should use glob pattern instead, there are several places we need to update manually  when we add a new plugin, including but might not limited to:

- pnpm-workspace.yaml
- webpack/config.ts
- .i18n-codegen.json
- plugin-infra/register.ts
- packages/plugin-infra/src/types.ts

We can save steps and reduce possible mistakes when add a new plugin.

Comment (Jack-Works) [reply pnpm-workspace.yaml:12]: I have opened a new issue for it. https://github.com/DimensionDev/Maskbook/issues/5666

**Pull request merged**